src/main/app/src/components/common/CookieModal.js

```diff
@@ -0,0 +1,262 @@
+    'z-index': '9999',
+    'background-color': 'white',
+    width: '60%',
+    border: '1px solid #ccc',
+    'box-shadow': '1px 1px 1px black',
+    padding: '16px',
+    left: '20%',
+    top: '5%',
+    'box-sizing': 'border-box',
+    'border-radius': '10px',
+    a: {
+      color: '#0a789c !important',
+      'margin-top': '5px',
+      'margin-bottom': '15px',
+      'text-decoration': 'none',
+    },
+    '@media (max-width:960px)': {
+      width: '90%',
+      left: '5%',
+    },
+  },
+
+  settings: {
+    'margin-top': '15px',
+    'margin-left': '2%',
+  },
+
+  settingsCheckbox: {
+    outline: '1px #3A7A10',
+    'box-shadow': 'none',
+    'font-size': '2em',
+    ':checked': {
+      'background-color': '#3A7A10',
+      icon: 'pointer',
+    },
+  },
+
+  settingsButton: {
+    'margin-top': '15px',
+    'margin-right': '5px',
+    height: '40px',
+    color: '#3A7A10',
+    'background-color': 'white',
+    padding: '4px 12px 4px',
+    cursor: 'pointer',
+    border: '2px solid transparent',
+    'border-color': '#3A7A10',
+    'border-radius': '4px',
+    'font-weight': '600',
+    float: 'right',
+    '&:hover': {
+      backgroundColor: '#e6ffc1',
+    },
+  },
+
+  acceptButton: {
+    'margin-top': '15px',
+    'margin-right': '5px',
+    height: '40px',
+    color: '#FFFFFF',
+    'background-color': '#3A7A10',
+    padding: '4px 12px 4px',
+    cursor: 'pointer',
+    border: '1px solid transparent',
+    'border-radius': '4px',
+    'border-color': '#3A7A10',
+    'font-weight': '600',
+    float: 'right',
+    '&:hover': {
+      backgroundColor: '#43a047',
+    },
+  },
+
+  textExpandLink: {
+    color: '#3A7A10',
+    'margin-top': '5px',
+    'margin-bottom': '15px',
+    'margin-left': '3%',
+    'font-weight': 'bold',
+  },
+});
+
+const CookieModal = (props) => {
+  const classes = useStyles();
+
+  const mandatoryCookieName = 'oph-mandatory-cookies-accepted';
+
+  const [settingsOpen, setSettingsOpen] = useState(false);
+  const [fullCookieInfoOpen, setFullCookieInfoOpen] = useState(false);
+  const [cookiesAlreadyAccepted, setCookiesAlreadyAccepted] = useState(
+    Cookies.get(mandatoryCookieName)
+  );
+
+  const [statisticCookiesAccepted, setStatisticCookiesAccepted] = useState(false);
+  const [marketingCookiesAccepted, setMarketingCookiesAccepted] = useState(false);
+
+  const { data, isLoading } = useContentful();
+
+  const single = (entry) => Object.values(entry || [])[0] || {};
+
+  const contentfulTexts = single(data.cookieModalText);
+
+  const fields = {
+    shortContent: contentfulTexts['shortContent'] || '',
+    fullContent: contentfulTexts['fullContent'] || '',
+    heading: contentfulTexts['heading'] || '',
+    expandLinkText: contentfulTexts['expandLinkText'] || '',
+    settingsButtonText: contentfulTexts['settingsButtonText'] || '',
+    settingsButtonCloseText: contentfulTexts['settingsButtonCloseText'] || '',
+    acceptButtonText: contentfulTexts['acceptButtonText'] || 'Hyväksy',
```

Comment:
Ei kovakoodattuja käännöksiä
'Hyväksy' -> t('yleiset.hyvaksy')

Reply:
Olin vähän epävarma miten nuo fallbackit ylipäänsä kannattaisi hoitaa, koska eihän tuolla modaalilla tee mitään ellei kaikkia tekstejä saada contentfulista. Vähän rumahan se kyllä on noin, oikeastaan laitoin tuohon vain jotain että sai testeissä erotettua napin muista.

Reply:
Hmm, pitäisköhän nää yksisanaiset perustekstit (Hyväksy / Sulje) tulla vaan käännöspalvelusta? Vois varmistaa. Noi pitkät litaniat varmaan pitää olla kustomoitavissa

Reply:
Siinä ois kyllä tavallaan järkeä, mutta olin ajatellut että ne voisi laittaa tulemaan samasta lähteestä kuin muunkin sisällön, koska tuo keksimodaali on kuitenkin selkeästi oma kokonaisuutensa jolle joka tapauksessa haetaan data contentfulista.

Reply:
Tehää näin, jos pistät noi fallbackit käännöspalvelusta näiden nappien osalta niin napit edes näkyy kunnolla jos tekstejä ei jostain syystä tulisikaan :+1: